bin/src/_main.rs

```diff
@@ -611,6 +585,38 @@ pub async fn _main(
     }
 }
 
+async fn set_up_leader(
+    namespace: &Option<String>,
+    k8s_client: &Kube_Client,
+    pod_name: &Option<String>,
+    feature: String,
+) -> FeatureLeaderMeta {
+    let lease_api =
+        k8s::lease::get_k8s_lease_api(&namespace.clone().unwrap_or_default(), k8s_client.clone())
+            .await;
+    let leader = FeatureLeader::new(
+        namespace.clone().unwrap_or_default(),
+        feature,
+        pod_name.clone().unwrap_or_default(),
+        lease_api,
+    );
+
+    let is_starting_leader = leader.try_claim_feature_leader().await;
+    let leader_lease = leader.get_feature_leader_lease().await;
+    let mut interval = DEFAULT_CHECK_FOR_LEADER_S;
+    if let Ok(lease) = leader_lease {
+        interval = lease
+            .spec
+            .map(|s| s.lease_duration_seconds.unwrap_or(0i32))
+            .unwrap_or(0i32)

```

Comment:
this hardcoded "0" is the same as DEFAULT_CHECK_FOR_LEADER_S?

Reply:
it is not, good eye. Going to rework this a bit.

Reply:
added a check where if this comes in 0, A the lease is misconfigured but it'll default back to DEFAULT_CHECK_FOR_LEADER_S